s3fs/tests/test_s3fs.py

```diff
@@ -52,6 +53,10 @@ def s3():
     client = boto3.client('s3')
     client.create_bucket(Bucket=test_bucket_name, ACL='public-read')
 
+    bucket = client.create_bucket(Bucket=versioned_bucket_name, ACL='public-read')
+    bucket_versioning = boto3.resource('s3').BucketVersioning(versioned_bucket_name)
```

Comment:
versioning is enabled on the bucket itself, not a dir

Reply:
A method that sets versioning for a given bucket, in that case? Or should that only be done via the console?

Reply:
Typically that would probably done by whoever administers the bucket.  Adding bucket versioning is something that should be done carefully due to the increased costs associated with storing additional copies of the data for different versions.

I don't think that bucket creation/management should be in scope for s3fs since modifying those kinds of parameters requires quite a high degree of IAM permissions.

Reply:
OK, fine to ignore that, then. Maybe someone will request it later.